### PR TITLE
[NAE-1766] Options on multichoice autocomplete do not refresh in a specific situation

### PR DESCRIPTION
**Pull request by machacjozef**
# Description

Fixes [NAE-1766]

## Dependencies

No new dependencies were introduced

### Third party dependencies

No new dependencies were introduced

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

This was tested manually.

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |      Linux Mint 20.3     |
| Runtime             |  Node v12.22.12         |
| Dependency Manager  |   NPM 6.14.16    |
| Framework version   |    Angular 13.3.1     |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @timbez      
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1766]: https://netgrif.atlassian.net/browse/NAE-1766?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: projects/netgrif-components-core/src/lib/data-fields/multichoice-field/multichoice-autocomplete-field/abstract-multichoice-autocomplete-field-component.component.ts [hunk 1]
```diff
@@ -24,20 +24,20 @@ export abstract class AbstractMultichoiceAutocompleteFieldComponentComponent imp
     filteredOptions: Observable<Array<MultichoiceFieldValue>>;
 
     ngOnInit() {
-        this.filteredOptions = this.formControlRef.valueChanges.pipe(
-            startWith(''),
-            map(value => this._filter(value).filter((option) => !this.multichoiceField.value.includes(option.key)))
-        );
+        this.formControlRef.valueChanges.subscribe(newVal => {
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/data-fields/multichoice-field/multichoice-autocomplete-field/abstract-multichoice-autocomplete-field-component.component.ts:27]: Subscription should be unsubscribed in onDestroy

File: projects/netgrif-components-core/src/lib/data-fields/multichoice-field/multichoice-autocomplete-field/abstract-multichoice-autocomplete-field-component.component.ts [hunk 2]
```diff
@@ -24,20 +24,20 @@ export abstract class AbstractMultichoiceAutocompleteFieldComponentComponent imp
...
+        })
     }
 
     ngOnDestroy(): void {
         this.filteredOptions = undefined;
     }
 
     add(event: MatChipInputEvent): void {
-        const value = (event['key'] || '').trim();
+        const value = (event['key'] || '');
```
Comment (minop) [inline projects/netgrif-components-core/src/lib/data-fields/multichoice-field/multichoice-autocomplete-field/abstract-multichoice-autocomplete-field-component.component.ts:37]: Are the keys only strings? If the keys were numbers then `0` would evaluate to `''` which is bad. Maybe `??` would be better than `||`.

**Review by minop — CHANGES_REQUESTED** [review]

File: projects/netgrif-components-core/src/lib/data-fields/multichoice-field/multichoice-autocomplete-field/abstract-multichoice-autocomplete-field-component.component.ts (hunk 1 above)
Comment (machacjozef) [reply projects/netgrif-components-core/src/lib/data-fields/multichoice-field/multichoice-autocomplete-field/abstract-multichoice-autocomplete-field-component.component.ts]: Done

File: projects/netgrif-components-core/src/lib/data-fields/multichoice-field/multichoice-autocomplete-field/abstract-multichoice-autocomplete-field-component.component.ts (hunk 2 above)
Comment (machacjozef) [reply projects/netgrif-components-core/src/lib/data-fields/multichoice-field/multichoice-autocomplete-field/abstract-multichoice-autocomplete-field-component.component.ts]: Done

**Review by minop — APPROVED** [review]

**Review by mladoniczky — APPROVED** [review]

**Pull request merged**